sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1718,7 +1718,7 @@ object SQLConf {
         "Data source V1 now silently drops partitionBy columns for non-file-format sources; " +
         "turning the flag on provides a way for these sources to see these partitionBy columns.")
       .booleanConf
-      .createWithDefault(false)
+      .createWithDefault(true)
```

Comment:
Actually, why do we need this configuration if it's not invasive?

Reply:
I think it was to avoid possible regression for the release 2.4.3, which was reasonable at that time.

Reply:
It's not intrusive then shouldn't be no regression strictly though. Yea but I got that it was an extra caution. Can we remove it now in master?

Reply:
Agree, we should be removing these in 3.0 rather than defaulting back to legacy behavior.

Reply:
Sounds good to me. I will go ahead and remove this feature flag altogether if no objections.

Reply:
+1

Reply:
cc @gatorsmile

Reply:
+1